File: backend/app/schemas/learning.py
from pydantic import BaseModel
from typing import List, Optional
import uuid

class QuizSchema(BaseModel):
    id: uuid.UUID
    lesson_id: uuid.UUID
    question: str
    options: List[str]
    correct_option_index: int
    
    class Config:
        from_attributes = True

class LessonSchema(BaseModel):
    id: uuid.UUID
    course_id: uuid.UUID
    title: str
    content: str
    order: int
    quizzes: List[QuizSchema] = []
    
    class Config:
        from_attributes = True

class CourseSchema(BaseModel):
    id: uuid.UUID
    title: str
    description: str
    level: str
    lessons: List[LessonSchema] = []
    
    class Config:
        from_attributes = True

class UserProgressSchema(BaseModel):
    lesson_id: uuid.UUID
    completed: bool
    score: int
    
    class Config:
        from_attributes = True

class UserProgressUpdate(BaseModel):
    user_id: uuid.UUID
    lesson_id: uuid.UUID
    score: int

File: frontend/src/api/learning.ts
import { apiClient, handleApiError } from './client';

export interface Quiz {
  id: string;
  lesson_id: string;
  question: string;
  options: string[];
  correct_option_index: number;
}

export interface Lesson {
  id: string;
  course_id: string;
  title: string;
  content: string;
  order: number;
  quizzes: Quiz[];
}

export interface Course {
  id: string;
  title: string;
  description: string;
  level: string;
  lessons: Lesson[];
}

export interface UserProgress {
  lesson_id: string;
  completed: boolean;
  score: number;
}

export const getCourses = async (): Promise<Course[]> => {
  try {
    const response = await apiClient.get('/learning/courses');
    return response.data;
  } catch (error) {
    throw new Error(handleApiError(error));
  }
};

export const getUserProgress = async (userId: string): Promise<UserProgress[]> => {
  try {
    const response = await apiClient.get(`/learning/progress/${userId}`);
    return response.data;
  } catch (error) {
    throw new Error(handleApiError(error));
  }
};

export const updateUserProgress = async (userId: string, lessonId: string, score: number): Promise<void> => {
  try {
    await apiClient.post('/learning/progress', {
      user_id: userId,
      lesson_id: lessonId,
      score: score
    });
  } catch (error) {
    throw new Error(handleApiError(error));
  }
};
